### DISCUSSION TITLE
How to install PyG with RAPIDS

### DISCUSSION THREAD

**Discussion by chester-tan** [pyg-team/pytorch_geometric#5927] [Q&A]
Hi! I've been trying to install a compatible version of cuda-enabled PyG with RAPIDS (cugraph, cudf, etc.) but struggling.

I've found that none of the supported binaries have matching cuda versions. I found a pytorch docker image that has a cuda version (=11.5) 
compatible with PyG but am unable to install RAPIDS in it due to conflicting dependencies: https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel_21-12.html#rel_21-12

Has anyone successfully created a python environment with both PyG on cuda and RAPIDS?

**Comment by rusty1s** [comment]
I don't have any experience with this, but happy to help. What are the conflicting dependencies?

**Reply by chester-tan** [reply]
Thanks for the kind reply! I just solved this yesterday but forgot to update -- got mamba to solve the conflicting dependencies in an ubuntu container (conda wasn't able to do this), after reading that rapids has "cuda enhanced compatibility" to support all 11.x cuda versions (<11.5).

My Dockerfile:
```
FROM ubuntu

RUN apt-get update && apt-get install -y wget 
RUN wget https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-Linux-x86_64.sh
RUN bash Mambaforge-Linux-x86_64.sh -b

RUN /root/mambaforge/bin/mamba init

RUN /root/mambaforge/bin/mamba install cuda -y -n base -c nvidia/label/cuda-11.3.1
RUN /root/mambaforge/bin/mamba install cudatoolkit==11.3.1 python rapids==22.10 pytorch torchvision torchaudio pyg modin pandas jupyter -y -n base -c rapidsai -c nvidia -c pytorch -c pyg -c conda-forge
```
